### PR TITLE
Resolve ILLink warnings in System.ComponentModel.TypeConverter (Final)

### PR DESCRIPTION
**Pull request by eerhardt**
Contributes to #45623

### CODE REVIEW

File: src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/MaskedTextProvider.cs [hunk 1]
```diff
@@ -19,6 +19,7 @@ namespace System.ComponentModel
     /// to the test string itself, since mask literals cannot be modified (i.e: replacing on a literal position
     /// will actually replace on the nearest edit position forward).
     /// </summary>
+    [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)]
```
Comment (vitek-karas) [inline src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/MaskedTextProvider.cs:22]: I know it's ... not polite... but I have to say that I'm supper happy with the linker feature for this. One-line fix for a gnarly problem.

Nit: I might add a comment describing where/how is this annotation needed - that's the one thing linker doesn't make obvious in this case.

**Review by vitek-karas — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/MaskedTextProvider.cs [hunk 2]
```diff
@@ -19,6 +19,7 @@ namespace System.ComponentModel
     /// to the test string itself, since mask literals cannot be modified (i.e: replacing on a literal position
     /// will actually replace on the nearest edit position forward).
     /// </summary>
+    [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] // needed by Clone() to preserve derived ctors
```
Comment (joperezr) [inline src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/MaskedTextProvider.cs:22]: I see your comment in there but I still don't fully follow. What does having a DynamicallyAccessedMembers attribute on a class mean? Looking at your code I would have expected Clone to instead have a DynamicDependency attribute?

**Review by joperezr — APPROVED** [review]
Added a question more for my understanding than anything else, but other than that this looks good.

File: src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/MaskedTextProvider.cs (hunk 2 above)
Comment (eerhardt) [reply src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/MaskedTextProvider.cs:22]: See https://github.com/mono/linker/pull/1929.

Maybe we should update the `DynamicallyAccessedMembersAttribute` docs to describe what it means when the attribute is applied to a class/interface/struct?

Comment (eerhardt) [reply src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/MaskedTextProvider.cs:22]: Opened https://github.com/dotnet/runtime/pull/51408 for this.

**Comment by eerhardt** [comment]
Libraries test failure is #48851
Installer test failures are #51372

**Pull request merged**